scanner/decisions/0003-scanner-image-names.md

```diff
@@ -0,0 +1,63 @@
+# 0003 - Scanner image names
+
+- **Author(s):** Ross Tannenbaum <rtannenb@redhat.com>
+- **Created:** 2023-08-22
+
+## Status
+
+Accepted.
+
+## Context
+
+The [current version of StackRox Scanner](https://github.com/stackrox/scanner) shipped with each StackRox release is a fork of [Clair v2](https://github.com/quay/clair/tree/v2.1.8) which is versioned and released separately from the rest of StackRox.
+The separate versioning process has proven to cause confusion in the past, so we re-tag each Scanner release with the associated StackRox release.
+For example, the Scanner 2.30.3 release triggered the creation of the following images:
+
+* scanner:2.30.3
+* scanner-slim:2.30.3
+* scanner-db:2.30.3
+* scanner-db-slim:2.30.3
+
+Upon cutting the related StackRox release, 4.1.3, the images were re-tagged as follows:
+
+* `scanner:4.1.3`
+* `scanner-slim:4.1.3`
+* `scanner-db:4.1.3`
+* `scanner-db-slim:4.1.3`
+
+There is an ongoing effort to overhaul the StackRox Scanner to align it more closely with [Clair v4](https://github.com/quay/clair).
+To align with Clair's versioning scheme, we refer to the new StackRox Scanner as StackRox Scanner v4.
+StackRox Scanner v4 is housed in this repository (as opposed to the previous, separate repository) and its release process is more tightly coupled with the StackRox release process.
+So, the Scanner v4 images will always be tagged the same as the [main image](https://quay.io/repository/stackrox-io/main).
+
+There will be a period of time when StackRox will ship both the older Scanner, Scanner v2, and the new Scanner, Scanner v4.
+It is clear that both versions of StackRox Scanner cannot have the same name and tag, so we must decide on a new name for either or both scanners.
+
+## Decision
+
+The new StackRox Scanner v4 images will be named as follows:
+
+* `scannerv4`
+* `scannerv4-db`
```

Comment:
No objection against `v4` (if Clair does not become `v5` itself), but I'd suggest to consider adding a dash

```suggestion
* `scanner-v4`
* `scanner-v4-db`
```

I don't have alternative suggestions at the moment; something like `scanner-ng` looks clumsy, IMO, and ScannerV4 seems to be an internal name in lots of places.

Since "V4" comes up in a bunch of user-visible places (like here - in image names), I'd suggest to add info to release notes and [ACS architecture diagrams](https://docs.openshift.com/acs/architecture/acs-architecture.html) about what V4 means so that users have right understanding (e.g. that `scanner-v4` is _newer_ technology than just-`scanner`, or e.g. in case ACS moves to major version 5.y.z and ScannerV4 image is called `scanner-v4:5.y.z`).